solr/core/src/java/org/apache/solr/handler/admin/api/SchemaNameAPI.java

```diff
@@ -46,7 +46,7 @@ public SchemaNameAPI(SolrCore solrCore) {
   }
 
   @GET
-  @Produces("application/json")
+  @Produces({"application/json", "application/xml", "application/javabin"})
```

Comment:
interesting....    And maybe going to make me sad in the future?    Do we actually want a javabin version for this API????

Reply:
Hah, yeah it does look weird as you add more and more formats to this list.  There must be a cleaner way to declare these.

> Do we actually want a javabin version for this API????

I think there's a good question of whether we want to hang on to javabin long term.  But while we do support it, I think we might as well allow it everywhere unless we have a reason not to?

Reply:
`application/javabin` is not a registered type, so should perhaps be something like `application/vnd.apachesolr.javabin`?

Reply:
Currently, v1 APIs return `application/octet-stream`, so I thought about putting that here for the sake of consistency, but that felt too generic.

I like your suggestion better Jan, will change.

Reply:
My suggestion is based on the recommendation by IANA, see https://en.wikipedia.org/wiki/Media_type#Vendor_tree. Here are the apache specific `vnd` formats I could find at https://www.iana.org/assignments/media-types/media-types.xhtml:
```
application/vnd.apache.arrow.file
application/vnd.apache.arrow.stream
application/vnd.apache.thrift.binary
application/vnd.apache.thrift.compact
```
So perhaps `application/vnd.apache.solr.javabin` is better.

Reply:
(pending @gerlowskija )

Reply:
Ah, yeah, this comment got buried but it's been on my list, sorry!

I'm 👍 on Jan's suggestion of `application/vnd.apache.solr.javabin`, I just need to implement it.  This will need changed both here, as well as in SolrJ so that it knows about the updated media-type value.  (There's not much v2 support in SolrJ, but you can manually construct your own request using the `V2Request` class)

Starting in on that now; will confirm here when that's finished.

Reply:
Turns out I had to fight SolrJ a bit to get this working.  I've pushed up a commit to get it working (see [here](https://github.com/apache/solr/pull/975/commits/565fb9cd3c37c5572d7ea9a4c9007dd623627c18) ), but I'm on the fence whether we should include it in this PR or whether it'd be better to stick with "application/octet-stream" for now and switch the mime type down the road.

The root of the issue is that the `getContentType()` method on SolrJ's `ResponseParser` interface returns a single string.  That is, ResponseParsers declare a single content-type that they can handle, and SolrJ will throw an error if Solr's response comes back with any other 'Content-Type' value.  This is a problem for us since we want BinaryResponseParser to support the new media type while maintaining compatibility with existing v1 and v2 requests that return `application/octet-stream`.

The linked commit works around this by adds a `ResponseParser.getContentTypes()` (note the plural) method for returning multiple content-types, deprecates the existing "single-string" `getContentType`, and updates the SolrClient usage to rely on the new method.